src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/ConvertToRecordHelpers.cs

```diff
@@ -54,14 +54,21 @@ internal static class CSharpOperationAnalysisHelpers
 
                     // otherwise we check to see which fields are compared (either by themselves or through properties)
                     var actualFields = GetEqualizedFields(methodBodyOperation, methodSymbol);
-                    return !actualFields.HasDuplicates(SymbolEqualityComparer.Default) &&
-                            actualFields.SetEquals(expectedComparedFields);
+                    return actualFields.SetEquals(expectedComparedFields);
                 }
             }
 
             return false;
         }
 
+        public static INamedTypeSymbol? GetIEquatableType(Compilation compilation, INamedTypeSymbol containingType)
+        {
+            // can't use nameof since it's generic and we need the type parameter
+            var equatableMetadataName = compilation.GetBestTypeByMetadataName("System.IEquatable`1")?.MetadataName;
```

Comment:
So we're getting the type only to go back to the metadata name -- that seems a bit fishy here. Is the real problem here the lack of an OriginalDefinition comparison in the lambda below?

Reply:
The problem is that the IEquatable symbol we are looking for has a generic type parameter which should be the type that we're currently in. However, we can't really find that type, or the IEquatable symbol with the type filled in (without looking through the interfaces themselves), so we use the metadata names, which are always the same regardless of the type arguments. We need to find the type and get it's metadata name back because it could be fully qualified or not, depending on the import status.

Reply:
Definitely open to alternative solutions here but I wasn't able to find anything that would get me the symbol with the type argument filled in with `containingType` or a way to compare a generic type with type arguments to its equivalent generic type without type arguments filled in yet.

Reply:
Switched to construct an `UnboundGenericType` for the `IEquatable` type and all the interfaces we compare with.